packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -68,11 +70,20 @@ export const withBlockBindingSupport = createHigherOrderComponent(
 				return;
 			}
 
+			const bindingsEntries = Object.entries( bindings );
 			const attributes = {};
 
-			for ( const [ attributeName, boundAttribute ] of Object.entries(
-				bindings
-			) ) {
+			if ( bindings[ DEFAULT_ATTRIBUTES ] !== undefined ) {
```

Comment:
As far as I understand, the https://github.com/WordPress/gutenberg/pull/60694#issuecomment-2126374085 from @SantosGuillamot is about the changes applied in the general processing of block bindings in the block editor. At the moment, the implementation proposed makes `__default` a keyword that would work with every possible type of block binding source on the client, but there is no matching logic on the server that would make it universally applicable. What alternatives would be possible to limit it to Pattern Overrides only?

Reply:
With the `getValue` implementation, I don't think there's any way to move this logic into pattern overrides source itself. At the moment this file checks each attribute individually so it expects the `metadata.bindings` attribute to use the same attribute name as the block attribute.

If it were `getValues` instead, I think it might be possible, as perhaps the code would encounter `__default` in the `metadata.bindings`, know that there's an active binding of some kind, and defer completely to the `getValues` implementation to return the matching values for each attribute. Within `getValues` the code could expand the `__default` binding to the complete supported attribute list

But probably the easiest implementation is to add some hard-coding in `useBindingAttributes` so that `__default` only works when it's set to `core/pattern-overrides`.

I realise that there's also ongoing efforts to refactor this into the store right now so the implementation might change.

Reply:
> But probably the easiest implementation is to add some hard-coding in useBindingAttributes so that __default only works when it's set to core/pattern-overrides.

Yes, that works for now 👍🏻

Reply:
I've made those changes. Also tried to make the changes to `useBindingAttributes` a lot more minimal so its easier to port to a different implementation, now the `__default` binding is handled by one function, and the rest of the code is the same.